omnigan/generator.py

```diff
@@ -1,14 +1,15 @@
+import torch
 import torch.nn as nn
 from omnigan.utils import init_weights
-
-# * https://docs.fast.ai/vision.models.unet.html
-# * Dynamic UNet FastAI
+from omnigan.blocks import Conv2dBlock, ResBlocks, SpadeDecoder, BaseDecoder
 
 # --------------------------------------------------------------------------
 # -----  For now no network structure, just project in a 64 x 32 x 32  -----
 # -----   latent space and decode to (3 or 1) x 256 x 256              -----
 # --------------------------------------------------------------------------
 
+# TODO think about how to use the classifier probs at inference
```

Comment:
I'll need you on this @adrienju: we've got to brainstorm about how to use the 1-h vectors specifying the target domain to the generator.

It could be that we don't actually need 2 different translation decoders, and probably not even an adaptation decoder as long as we use the "bit-flip" trick. This would mean that with only 1 decoder we can target the 4 domains if we condition the SPADE blocks on   1-hot encoded (or soft distributed with a softmax) feature maps:

vector (size [4,]) * ones (size [4, 256, 256])

makes sense? To be discussed anyway